src/Microsoft.Extensions.Configuration.Ini/IniConfigurationExtensions.cs

```diff
@@ -75,8 +75,12 @@ public static IConfigurationBuilder AddIniFile(this IConfigurationBuilder builde
 
             if (provider == null && Path.IsPathRooted(path))
             {
-                provider = new PhysicalFileProvider(Path.GetDirectoryName(path));
-                path = Path.GetFileName(path);
+                string pathToFile;
+                provider = FileConfigurationProvider.ResolvePhysicalFileProvider(path, out pathToFile);
```

Comment:
I talked to @victorhurdugaci briefly and it seems that solving this only if an absolute path is specified should be enough to address the Antares scenario he is hitting. However it seems that we could setup ourselves better to address additional scenarios (e.g. relative paths and path traversal as described in #464) if we made this new API a bit more general. 

E.g. we can make the method `ResolvePhysicalFileProvider(ref IFileProvider provider, ref string path)` and move the whole `if (provider == null && Path.IsPathRooted(path))` block inside it. Then the additional improvements mentioned above shouldn't require changes to the individual providers or the API surface.